libs/net/dal/Services/ReportService.cs

```diff
@@ -25,11 +43,41 @@ public IEnumerable<Report> FindAll()
             .OrderBy(a => a.SortOrder).ThenBy(a => a.Name).ToArray();
     }
 
+    /// <summary>
+    /// Find the report for the specified 'id'.
+    /// </summary>
+    /// <param name="id"></param>
+    /// <returns></returns>
     public override Report? FindById(int id)
     {
         return this.Context.Reports
             .Include(n => n.SubscribersManyToMany).ThenInclude(s => s.User)
             .FirstOrDefault(n => n.Id == id);
     }
+
+    /// <summary>
+    /// Make a request to Elasticsearch to find content for the specified 'report'.
+    /// </summary>
+    /// <param name="report"></param>
+    /// <returns></returns>
+    /// <exception cref="Exception"></exception>
+    public async Task<IEnumerable<Content>> FindContentWithElasticsearchAsync(Report report)
+    {
+        var response = await _client.SearchAsync<Content>(s =>
+        {
+            var result = s
+                .Pretty()
+                .Index(_elasticOptions.PublishedIndex) // TODO: Switch to unpublished if the report settings require it.
+                .Query(q => q.Raw(report.Filter.ToJson()));
```

Comment:
Raw JSON request.  @CelloLiu @jdtoombs This is what I want to do with the `ContentService` function too at some point.  Then we don't have to convert the `ContentFilter` into an elastic query.  However, it will mean that we need to submit an elastic query from the frontend instead.

Reply:
I like this